packages/core/src/session_pool/session.ts

```diff
@@ -268,7 +264,8 @@ export class Session {
      * @returns Whether the session was retired.
      */
     retireOnBlockedStatusCodes(statusCode: number, blockedStatusCodes: number[] = []): boolean {
-        const isBlocked = STATUS_CODES_BLOCKED.concat(blockedStatusCodes).includes(statusCode);
+        // @ts-expect-error
+        const isBlocked = this.sessionPool.blockedStatusCodes.concat(blockedStatusCodes).includes(statusCode);
         if (isBlocked) {
             this.retire();
```

Comment:
Let's make a note to revisit this later. This function sucks. It's not clear whether it retires on the blocked codes you pass or on some other codes. We can't change it now, because it would be breaking, but let's not forget about it 😄

Reply:
Maybe lets deprecated the second parameter? Given its now configurable on the `SessionPool`.